### ISSUE TITLE
browser-security/no-innerhtml

### ISSUE THREAD

**Issue by alexgagnon** [ofri-peretz/eslint#1056]
### Rule Name

browser-security/no-innerhtml

### Code Snippet

```typescript
const example = await sanitize('');
```

### Why is this safe?

'sanitize' is a custom function we have that is an async wrapper around DOMPurify. This should be allowed as 'sanitize' is on the default list of trustedSanitizers. Due to the async nature, it throws. Just removing the 'await' keyword, which makes the code invalid, makes the rule pass.

### Environment

- **Package Version**: 2.1.5
- **ESLint Version**: 9.39.5

**Comment by ofri-peretz** [comment]
Thanks for the detailed report — this was a real bug, and your repro pinpointed it exactly.

Root cause: `isSanitized()` (the check behind `trustedSanitizers`) only recognized a trusted sanitizer when the value being judged was itself a call expression. `await sanitize(x)` is an `AwaitExpression` wrapping that call, and that wrapper was never unwrapped before the check — so an async sanitizer wrapper on the default allowlist (like your `sanitize`) still reported as unsanitized once a caller `await`ed it, exactly as you found: removing `await` (even though it breaks the code) made the rule pass, because then the rule saw a plain `CallExpression` again.

Fix is up in #1070: it unwraps `AwaitExpression` the same way the rule already unwraps `ChainExpression` (for `DOMPurify?.sanitize(x)`), then judges the call underneath. Added regression tests for both `await sanitize(x)` and `await DOMPurify.sanitize(x)` so this can't silently regress.

Will leave this issue open until the fix lands on `main` and a release goes out — I'll update here when that happens.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ofri-peretz** [comment]
The fix has merged (#1070) — `await sanitize(x)` and `await DOMPurify.sanitize(x)` are now correctly recognized as trusted sanitizer calls. It'll be included in the next release of `eslint-plugin-browser-security`.

---
_Generated by [Claude Code](https://claude.ai/code)_
